tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -370,6 +370,18 @@ def caseNumberNotEmpty(start=0, end=None):
     )
 
 
+def calendarQuarterIsValid(start=0, end=None):
+    """Validate that the calendar quarter value is valid."""
+    return make_validator(
+        lambda value: value[start:end].isnumeric() and int(value[start:end - 1]) >= 2020
+        and int(value[end - 1:end]) > 0 and int(value[end - 1:end]) < 5,
+        lambda value,
+        row_schema,
+        friendly_name,
+        item_num: f"{row_schema.record_type}: {value[start:end]} is an invalid value for the CALENDAR_QUARTER field.",
```

Comment:
@reitermb, @victoriaatraft, can I get some UX eyes on this to verify it is reasonable?

Reply:
@elipe17 how would they select an invalid value and what values should they select? 
ie: are they selected a month, date or quarter that should be not be available? And they should only select a value that happened within x time frame?

Reply:
The `CALENDAR_QUARTER` field is of the form: `YearQuarter` eg `20204`. An invalid value could be anything from an invalid year (year is not a numeric, year is not >= 2020) to an invalid quarter (quarter is not a numeric, quarter is not on the range [1, 4]). A valid value would be `20204` and some invalid values would be: `20194`, `202!4`, `2020 `, etc... Does that help?

Reply:
Thank you. Perhaps: 
` item_num: f"{row_schema.record_type}: {value[start:end]} is invalid and must be a valid numerical value with no extra spaces for the CALENDAR_QUARTER field."
`

I believe this was also pending @reitermb input which we should get today. I'll update my comment once that's been acknowledged.

Reply:
I like expanding on what it means by invalid, I'd suggest we get even more specific if we lean toward expanding the message, e.g. ...`[string] is invalid. Calendar Quarter must be a numeric representing the Calendar Year and Quarter formatted as YYYYQ`.

Thoughts there @elipe17 @victoriaatraft ?

Does that get us into trouble re: validators needing to be generic because we're using this one in other places?

Reply:
No trouble at all. This message is tied to a specific calendar quarter cat1 validator and will not suffer from a loss of generality. I will update the message following your guidance!